packages/millicast-sdk/src/utils/Diagnostics.js

```diff
@@ -26,10 +27,10 @@ const setFeedId = (feedId) => { diagnostics.feedId = feedId }
 const setConnectionState = (connectionState) => { diagnostics.connection = connectionState }
 
 const setStats = (stats) => {
-  if (diagnostics.stats.length === MAX_STATS_HISTORY_SIZE) {
-    diagnostics.stats.shift()
+  if (tempStats.length === MAX_STATS_HISTORY_SIZE) {
```

Comment:
```suggestion
  if (tempStats.length === Math.min(statsCount, MAX_STATS_HISTORY_SIZE)) {
```

Reply:
`statsCount` cannot be access from here. This is a method called from the PeerConnectionStats interval on the `initStats` function. The temp array seems to be the only way to go.

Reply:
OK, I got it. `statsCount` is only for `getStats` function. e.g. Although we could record 5 at most, user still could get 3 of the latest ones. You can ignore that suggestion. But I am still thinking `diagnostics.stats` should work, isn't it?

Reply:
I thought it did work, but some times, the interval were the stats are being saved is called during the getDiagnostics function and it overrides any slice or change done before